.eslintrc.js

```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  extends: 'standard',
+  extends: 'standard-with-typescript',
```

Comment:
**question** I'm not sure that does what we hope it does. Looking at [the config it brings], it looks like it only adds TypeScript parsing for `.ts` and `.tsx` files. 

This won't match anything on our project, and will not add type information for the linter while parsing our JS files. 

Instead, don't we want to keep `standard` but add the `@typescript-eslint` plugin and set `@typescript-eslint/parser` to parse the files that will get bundled? 

I gave that a quick shot and ended with the following errors when linting ran:

```txt
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting##i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file
```

Reply:
Ah sorry @romaricpascal you're right we're not using `@typescript-eslint/parser`

But we are using the TypeScript ESLint plugin that comes with `standard-with-typescript`:

```js
plugins: ['@typescript-eslint']
```

We may want to opt in to the parser in future but likely still needs:

* https://github.com/alphagov/govuk-frontend/issues/2886

The ESLint TypeScript parser likes to have clear boundaries between child projects
https://typescript-eslint.io/linting/typed-linting/monorepos/

Thanks for taking a close look

Reply:
>Instead, don't we want to keep standard but add the @typescript-eslint plugin and set @typescript-eslint/parser to parse the files that will get bundled? Or [maybe follow @typescript-eslint's recommended setup](https://typescript-eslint.io/getting-started/#step-2-configuration)

@romaricpascal Pushed again, now following the recommended setup 👍

Also useful for prototype checks in your PR too:

* https://github.com/alphagov/govuk-frontend/pull/3153